### PR TITLE
check remote host being available.

### PR DESCRIPTION
**Pull request by heejaechang**
when host is shutting down it is possible GetRemoteHostClient return null even though host itself support remote hosting.

### CODE REVIEW

**Comment by heejaechang** [comment]
@CyrusNajmabadi can you take a look?

**Comment by heejaechang** [comment]
fix #17768

File: src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngineFactory.cs [hunk 1]
```diff
@@ -76,7 +76,11 @@ private class RemoteUpdateEngine : ISymbolSearchUpdateEngine
                     _sessionDoNotAccessDirectly = null;
 
                     _client = await _workspace.GetRemoteHostClientAsync(CancellationToken.None).ConfigureAwait(false);
-                    _client.ConnectionChanged += OnConnectionChanged;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngineFactory.cs]: are there other people that might observe that the client returned by GetRemoteHostClientAsync could be null?

**Comment by heejaechang** [comment]
sure

**Comment by heejaechang** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by heejaechang** [comment]
tagging @KevinH-MS I renamed GetRemoteHostClientAsync to TryGetRemoteHostClientAsync

**Comment by heejaechang** [comment]
retest windows_release_vs-integration_prtest  please

**Pull request merged**